mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -72,6 +72,22 @@ private[regression] trait LinearRegressionParams extends PredictorParams
 }
 
 /**
+ * Params for linear regression.
+ */
+private[regression] trait LinearRegressionParams extends LinearRegressionModelParams
```

Comment:
The refactor here is not only for this PR, since model usually has less params than estimator, we make _EstimatorParams_ extends from _ModelParams_ in other places, for example, _ALSParams_ extends _ALSModelParams_. I think the class hierarchy should be more clear.